src/main/java/steps/CartPageSteps.java

```diff
@@ -16,4 +19,13 @@ public static void goToCheckout() {
         cartPage = new CartPage();
         cartPage.clickOnFinalCheckoutButton();
     }
+
+    public static List<ProductModel> getProductsFromShoppingTable() {
+        cartPage = new CartPage();
+        return cartPage.shoppingCartTable.getProducts();
+    }
+
+    public static void validateIfProductIsInTable(ProductModel product, List<ProductModel> products) {
+        Assert.assertNotNull(products.stream().filter(p -> p.getName().equals(product.getName())).findFirst().orElse(null));
```

Comment:
Why not null assertion? You can just assert that list of products contain given name :)

Reply:
Outdated. The method has been changed so that it checks whether only ONE product has occurred in the list. Because more than one means an bug.
```java
    public static void checkIfProductAppearsOnce(ProductModel product, List<ProductModel> products) {
        var count = products
                .stream()
                .filter(p -> p.equals(product))
                .count();

        Assert.assertEquals(1, count);
    }
```